Ignore whitespace when rendering conditional sequence

dennisschagt opened the pull request:
[Idea](https://github.com/newsboat/newsboat/pull/868#issuecomment-610831063) by @tsipinakis:
> How about ignoring whitespace in conditionals instead, it's technically 'empty'? IMO I'd expect an all whitespace string to be considered empty.

Replaces PR #868.

Also adding format specifiers for unread/deleted/flags to have more control over formatting (`%f` combined some flags [using gen_flags](https://github.com/newsboat/newsboat/blob/09ec32244dca3f5699cb736636c8b5a2105d31ef/src/itemlistformaction.cpp#L1374-L1390).

Minoru commented on doc/newsboat.asciidoc:
```diff
@@ -1022,7 +1022,10 @@ by " ") and "download error" (indicated by "x").
 [[articlelist-format-a]]<<articlelist-format-a,+a+>>:Article author
 [[articlelist-format-D]]<<articlelist-format-D,+D+>>:Publication date. This can be tweaked further with <<datetime-format,+datetime-format+>>
-[[articlelist-format-f]]<<articlelist-format-f,+f+>>:Article flags
+[[articlelist-format-f]]<<articlelist-format-f,+f+>>:Combination of unread/deleted fields and article flags
```
The doc is not entirely correct here about the article flags: they're replaced by an exclamation mark or a space (see `gen_flags()`). I'd change this to:

> Two characters: 1) "N" if article is unread, "D" if article is deleted, a space otherwise; 2) "!" if article has flags, a space otherwise.

dennisschagt replied:
Good suggestion. Thanks!

Minoru commented on doc/newsboat.asciidoc:
```diff
@@ -1022,7 +1022,10 @@ by " ") and "download error" (indicated by "x").
+[[articlelist-format-f]]<<articlelist-format-f,+f+>>:Combination of unread/deleted fields and article flags
+[[articlelist-format-n]]<<articlelist-format-n,+n+>>:"unread" field
+[[articlelist-format-d]]<<articlelist-format-d,+d+>>:"deleted" field
+[[articlelist-format-F]]<<articlelist-format-F,+F+>>:Article flags
```
Please mention these three in the CHANGELOG.

Minoru commented on rust/libnewsboat/src/fmtstrformatter/mod.rs:
```diff
@@ -131,7 +131,7 @@ impl FmtStrFormatter {
     ) {
         match self.fmts.get(&cond) {
-            Some(value) if !value.is_empty() => {
+            Some(value) if !value.trim().is_empty() => {
```
This has to be tested and documented:
1) please add a test (the same one for Rust and C++);
2) please mention this change in the CHANGELOG;
3) add a note about this somewhere in the docs.

dennisschagt replied:
Done, done, and done.

dennisschagt commented:
> Please rebase these changes onto current master (aefe1d3) — I fixed CI breakage there.

Thanks for fixing those issues!
Rebased on current master.

Minoru commented:
Thanks, @dennisschagt!
